x/distribution/client/cli/query.go

```diff
@@ -0,0 +1,212 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	distr "github.com/cosmos/cosmos-sdk/x/distribution"
+)
+
+// GetCmdQueryParams implements the query params command.
+func GetCmdQueryParams(queryRoute string, cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "params",
+		Args:  cobra.ExactArgs(0),
+		Short: "Query distribution params",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			res, err := queryParams(cliCtx, cdc, queryRoute)
+			if err != nil {
+				return err
+			}
+
+			fmt.Println(string(res))
+			return nil
+
+		},
+	}
+	return cmd
+}
+
+func queryParams(cliCtx context.CLIContext, cdc *codec.Codec, queryRoute string) ([]byte, error) {
+	retCommunityTax, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/params/community_tax", queryRoute), []byte{})
+	if err != nil {
+		return nil, err
+	}
+
+	retBaseProposerReward, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/params/base_proposer_reward", queryRoute), []byte{})
+	if err != nil {
+		return nil, err
+	}
+
+	retBonusProposerReward, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/params/bonus_proposer_reward", queryRoute), []byte{})
+	if err != nil {
+		return nil, err
+	}
+
+	retWithdrawAddrEnabled, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/params/withdraw_addr_enabled", queryRoute), []byte{})
+	if err != nil {
+		return nil, err
+	}
+
+	return codec.MarshalJSONIndent(cdc, struct {
```

Comment:
can we define the struct and a `New...` function out of this scope to only have the its returned value here ?

Reply:
Sure, done.